.github/workflows/main.yml

```diff
@@ -53,17 +62,23 @@ jobs:
   # Check Code style quickly by running `rustfmt` over all code
   rustfmt:
     runs-on: ubuntu-latest
+    if: "github.repository == 'rust-lang/cargo'"
     steps:
     - uses: actions/checkout@v4
+      with:
+        fetch-depth: 2
     - run: rustup update stable && rustup default stable
     - run: rustup component add rustfmt
     - run: cargo fmt --all --check
 
   # Ensure there are no clippy warnings
   clippy:
     runs-on: ubuntu-latest
+    if: "github.repository == 'rust-lang/cargo'"
```

Comment:
How much do we need this if we block push branches?

This can be annoying for people intentionally forking, like a company recently did for hack week.

Reply:
Strictly speaking, it's not required. I mainly followed here what rust-lang/rust does.

But if you think this will actively hurt people, I'm open to removing it.

Reply:
I frequently run CI in my fork to test things. This would make it more difficult since I would need to comment this out.

Reply:
*I looked at your fork and didn't see any relevant PRs, so I'm assuming you don't use PRs to your own fork.*

Wouldn't you be also affected by the allowlist of branches? Or are you planning on making PRs to your own fork?

Reply:
The company never opened a PR against our repo. They all worked against a specific fork.

Reply:
> The company never opened a PR against our repo. They all worked against a specific fork.

Okay, then removing those conditions seems better. Removed.